tests/mechanic/launcher_test.py

```diff
@@ -115,23 +118,60 @@ def wait(self):
         return 0
 
 
+class MockProcess:
+    def __init__(self, pid):
+        self.pid = pid
+        self.killed = False
+
+    def name(self):
+        return "p{pid}".format(pid=self.pid)
+
+    def wait(self, timeout=None):
+        raise psutil.TimeoutExpired(timeout)
+
+    def kill(self):
+        self.killed = True
+
+
+def get_metrics_store(cfg):
+    cfg.add(config.Scope.application, "track", "params", None)
+    cfg.add(config.Scope.application, "system", "env.name", None)
+
+    ms = InMemoryMetricsStore(cfg)
+    ms.open(trial_id=str(uuid.uuid4()),
+            trial_timestamp=datetime.now(),
+            track_name="test",
+            challenge_name="test",
+            car_name="test")
+    return ms
+
+
 MOCK_PID_VALUE = 1234
 
 
 class ProcessLauncherTests(TestCase):
     @mock.patch('os.kill')
-    @mock.patch('subprocess.Popen',new=MockPopen)
+    @mock.patch('subprocess.Popen', new=MockPopen)
     @mock.patch('esrally.mechanic.java_resolver.java_home', return_value=(12, "/java_home/"))
     @mock.patch('esrally.utils.jvm.supports_option', return_value=True)
+    @mock.patch('esrally.utils.io.get_size')
     @mock.patch('os.chdir')
-    @mock.patch('esrally.config.Config')
-    @mock.patch('esrally.metrics.MetricsStore')
-    @mock.patch('esrally.mechanic.provisioner.NodeConfiguration')
     @mock.patch('esrally.mechanic.launcher.wait_for_pidfile', return_value=MOCK_PID_VALUE)
-    @mock.patch('psutil.Process')
-    def test_daemon_start_stop(self, process, wait_for_pidfile, node_config, ms, cfg, chdir, supports, java_home, kill):
+    @mock.patch('psutil.Process', new=MockProcess)
+    def test_daemon_start_stop(self, wait_for_pidfile, chdir, get_size, supports, java_home, kill):
+        cfg = config.Config()
+        cfg.add(config.Scope.application, "node", "root.dir", "test")
+        cfg.add(config.Scope.application, "mechanic", "keep.running", False)
+        cfg.add(config.Scope.application, "mechanic", "telemetry.devices", [])
+        cfg.add(config.Scope.application, "mechanic", "telemetry.params", None)
+
+        ms = get_metrics_store(cfg)
         proc_launcher = launcher.ProcessLauncher(cfg, ms, paths.races_root(cfg))
 
+        tmpdir = tempfile.mkdtemp()
```

Comment:
Is this really needed? As far as I can see all file system operations are mocked so we can just provide a fake path as string here?